src/ui/components/layouts/base_layout.rs

```diff
@@ -2,28 +2,49 @@ use crate::{
   contexts::theme_resource_context::ThemeResource,
   ui::components::{
     common::unpack::Unpack,
-    layouts::base_layout::{bottom_nav::BottomNav, top_nav::TopNav},
+    layouts::base_layout::{mobile_nav::MobileNav, side_nav::SideNav, top_nav::TopNav},
   },
 };
 use leptos::*;
 use leptos_meta::Html;
 use leptos_router::Outlet;
 
-mod bottom_nav;
+mod mobile_nav;
+mod side_nav;
 mod top_nav;
 
 #[component]
 pub fn BaseLayout() -> impl IntoView {
   let theme = expect_context::<ThemeResource>();
 
   view! {
-    <Transition>
-      <Unpack item=theme let:theme>
-        <Html attr:data-theme=theme/>
-        <TopNav/>
-        <Outlet/>
-        <BottomNav/>
-      </Unpack>
-    </Transition>
+    <div class="drawer h-full">
+      <input id="mobile-drawer" type="checkbox" class="drawer-toggle"/>
+      <div class="drawer-content h-full min-h-screen md:min-h-0 md:max-h-screen flex flex-col">
+        <Transition>
+          <Unpack item=theme let:theme>
+            <Html attr:data-theme=theme class="h-full max-h-screen overflow-y-hidden"/>
+            <TopNav/>
+            <div class="flex gap-x-4 h-full overflow-y-auto md:overflow-y-hidden grow">
+              <aside class="w-fit px-3.5 whitespace-nowrap pb-5 pt-3 border-e border-neutral hidden md:flex flex-col bg-base-300 overflow-y-auto h-full">
+
+                <SideNav/>
+              </aside>
+              <div class="grow h-fit md:h-auto">
+                <Outlet/>
+              </div>
+            </div>
+          </Unpack>
+          <MobileNav/>
+        </Transition>
+      </div>
+      <div class="drawer-side" class="md:hidden">
+        <label for="mobile-drawer" aria-label="Close mobile drawer" class="drawer-overlay"></label>
+        <aside class="w-2/3 whitespace-nowrap px-3.5 pb-5 pt-3 flex flex-col bg-base-300 overflow-y-auto h-full border-e border-neutral">
+
+          <SideNav/>
```

Comment:
Will have to tailor the mobile drawer menu more towards mobile in a later PR. Currently using the side nav as a placeholder.

Reply:
Hrm, yep, the responsive components of [drawer](https://daisyui.com/components/drawer/) are all they offer rn. Should be fine, and if there's anything else we need we can open up issues there.

Reply:
We can lively use the popover API for this in the future. From what I've seen of it, it's behavior is still somewhat inconsistent between browers, but it already has 86% support listed on caniuse.